cmd/dockerd/daemon.go

```diff
@@ -227,6 +232,28 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 	// Notify that the API is active, but before daemon is set up.
 	preNotifyReady()
 
+	const otelServiceNameEnv = "OTEL_SERVICE_NAME"
+	if _, ok := os.LookupEnv(otelServiceNameEnv); !ok {
+		id, err := daemon.LoadOrCreateID(cli.Config.Root)
```

Comment:
Yeah, that's kinda ugly. So I _think_ the motivation here may have been because @cpuguy83 ran OTEL on the integration tests, in which case there's "many" instances of `dockerd` (and sometimes multiple used in the same test).

Other cases could also mean if you collect OTEL centrally, but collect it for multiple hosts; I suggested to use the daemon "ID" for this, because that's an ID that's already visible to the user (part of `docker info` output).

But I agree that it's just "noise" if you only have a single daemon.

I think this would be fine to tweak in a follow-up, but we need to think what the best approach would be;

- default to `dockerd` (no ID after)
- perhaps allow `OTEL_SERVICE_NAME` to take a "template" (`dockerd-{{.ID}}`, `dockerd-{{.HostName}}`, etc..) and we could fill in that template.

🤔

Reply:
https://github.com/moby/moby/pull/45652#discussion_r1272545429 😄 

I think what would work is adding a span id to the daemon logs (when available) like bklog does would work to differentiate daemons.